src/sobject/sobject.cpp

```diff
@@ -79,15 +101,43 @@ sobject_ptr sobject_manager::create_object(const std::string& type, const std::s
     return obj;
 }
 
-void sobject_manager::add_object_type(const std::string& parent_type, const std::string& child_type) {
+void sobject_manager::remove_object(const std::string& name) {
     std::unique_lock lock{m_mtx};
-    m_object_types[parent_type].insert(child_type);
+    m_object_store.erase(name);
+    m_object_types[name].clear();
+    LOGINFO("Removed status object name={}", name);
+}
+
+bool sobject_manager::add_object_type(const std::string& parent_type, const std::string& child_type) {
+    bool res = false;
+    std::unique_lock lock{m_mtx};
+    auto p_map = m_object_types[parent_type].find(child_type);
+    if (p_map == m_object_types[parent_type].end()) {
+        m_object_types[parent_type][child_type] = 1;
+        res = true;
+    } else {
+        ++m_object_types[parent_type][child_type];
+    }
+    return res;
+}
+
+bool sobject_manager::remove_object_type(const std::string& parent_type, const std::string& child_type) {
+    bool res = false;
+    std::unique_lock lock{m_mtx};
+    auto count = m_object_types[parent_type][child_type];
```

Comment:
why do you need a type for each child type?

Reply:
m_object_types keeps track of types of children. sobject_manager has been designed that each child can have a separate type. So when a child is added its type has been added  m_object_types of parent list. Previously, it was only add child, so that the child type has been added to a set. Now that we want to remove a child, it is needed to check if other children have this removing child or not. If not, the child type has to be removed o.w we keep the type for the sake of other children type. Hence, it necessitates to keep track of count of a child_type.

Reply:
Can you give one example that under same parent, multiple child will be added with same child_type?

Reply:
Right now logstore family, all children have the same type logstore
parent type : LogStoreFamily
children type: HomeLogStore

Reply:
It is creating with type HomeLogStore_StoreId, every type is unique.

https://github.com/eBay/HomeStore/blob/stable/v3.x/src/homelogstore/log_store.cpp#L48

Reply:
HomeLogStore_${StoreId} is the name but they have the same type `HomeLogStore`